ARCs/arc-0041.md

```diff
@@ -0,0 +1,155 @@
+---
+arc: 41
+title: NFT's vault of NFTs & FTs
+description: Application ID storing NFTs & FTs on behalf of user
+author: Stéphane BARROSO (@SudoWeezy)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/213
+status: Draft
+type: Standards Track
+category: Interface
+created: 2023-06-16
+---
+
+## Abstract
+This standard is inspired by <a href="https://eips.ethereum.org/EIPS/eip-6551">EIP-6551</a>
+The goal is to create a smart contract who will give is owner a way to bundle multiple Algo/Fungible Token/NFTs at once.
+
+
+## Specification
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+### Glossary
+In the document we will use following syntax:
+- Authority NFT -> NFT that detain & manage others NFTs & FTs 
+- Vault -> Application ID holding NFTs & FTs of the Authority NFT
+
+### Events
+In order to allow wallet to identify the vault of the Authority NFT, an [ARC-28](./arc-0028.md) event must emitted when creating the vault.
+```json
+{
+  "events": [
+    {
+      "name": "arc41_AccountCreated",
+      "desc": "Creation of a Vault",
+      "args": [
+        {
+          "type": "address",
+          "name": "account",
+          "desc": "The current owner of the authority NFT"
+        },
+        {
+          "type": "uint64",
+          "name": "asaId",
+          "desc": "The asaId owner of the authority NFT, (0 if application NFT)"
+        },
+        {
+          "type": "uint256",
+          "name": "tokenId",
+          "desc": "The ID of the authority NFT, (0 if ASA)"
+        },
+        {
+          "type": "uint64",
+          "name": "AppId",
+          "desc": "The Application ID of the authority NFT, (0 if ASA)"
+        }
+      ]
+    }
+}
+```
+
+A `arc41_AccountCreated` event SHOULD be emitted, when a vault is created.
```

Comment:
This states 'should' be emitted, but line 27 says 'must' - which is correct?

Reply:
Thanks, it's a must